src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Notifications/Services/EmailHistoryService.cs

```diff
@@ -0,0 +1,24 @@
+﻿using AirBnB.Application.Common.Notifications.Services;
+using AirBnB.Persistence.Repositories.Interfaces;
+using FluentValidation;
+using System.Linq.Expressions;
+
+namespace AirBnB.Infrastructure.Common.Notifications.Services;
+
+public class EmailHistoryService(
+    IEmailHistoryRepository emailHistoryRepository,
+    IValidator<EmailHistory> emailHistoryValidator) : IEmailHistoryService
+{
+    public IQueryable<EmailHistory> Get(Expression<Func<EmailHistory, bool>>? predicate = null, bool asNoTracking = false)
+        => emailHistoryRepository.Get(predicate, asNoTracking);
+
+    public ValueTask<EmailHistory> CreateAsync(EmailHistory emailHistory, bool saveChanges = true, CancellationToken cancellationToken = default)
+    {
+        var validationResult = emailHistoryValidator.Validate(emailHistory);
```

Comment:
Use validation with IncludeRuleSets to specify the rule set to avoid the usage of default validation.